components/script/dom/userscripts.rs

```diff
@@ -40,13 +40,11 @@ pub fn load_script(head: &HTMLHeadElement) {
         files.sort();
 
         for file in files {
-            let name = match file.into_os_string().into_string() {
-                Ok(ref s) if s.ends_with(".js") => "file://".to_owned() + &s[..],
-                _ => continue
-            };
-            let new_script = doc.CreateElement(DOMString::from("script")).unwrap();
-            new_script.set_string_attribute(&local_name!("src"), DOMString::from(name));
-            node.InsertBefore(new_script.upcast(), first_child.r()).unwrap();
+            let mut f = File::open(&file).unwrap();
+            let mut contents = vec![];
+            f.read_to_end(&mut contents).unwrap();
+            let script_text = String::from_utf8(contents).unwrap();
```

Comment:
This should use `String::from_utf8_lossy`, to replace UTF-8 errors with U+FFFD instead of panicking. This show that Firefox does so for external scripts:

```html
data:text/html,<script src='data:text/html;charset=utf-8,document.write("%2580".charCodeAt(0).toString(16))'></script>
```
```
fffd
```